src/Core/Exception/ModuleException.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Exception;
+
+use Exception;
+
+class ModuleException extends Exception
+{
+    /**
+     * @var string[]
+     */
+    private $errors;
+
+    public function __construct(array $errors, $message = "", $code = 0, $previous = null)
+    {
+        $this->setErrors($errors);
+        parent::__construct($message, $code, $previous);
+    }
+
+    /**
+     * @param string[] $errors
+     *
+     * @return ModuleException
+     */
+    public static function buildFromArray(array $errors)
+    {
+        return new self($errors);
```

Comment:
Although I'm not sure this static factory is useful, is it?

Reply:
For me it also seems quite a strange way to do it. @matks, wdyt? You have suggested this approach I think the reason was to make module developers a simplified version of how to raise an exception?

Reply:
Because I wanted to allow module developers to build a `ModuleException` with either 1 or multiple error messages for maximum flexibility (with a constructor `public function __construct($error, ...)`

I expect most module developers to throw one exception for each error which means
```
throw ModuleException($error);
```
so `buildFromArray()` was a helper function to be able to deal with an array of errors

Reply:
Also we might consider adding a `addError()` function for developers which want to build the ModuleException gradually and throw it at the end 🤔 ? or is it too much ?